beryllia/__init__.py

```diff
@@ -109,6 +111,43 @@ async def _is_oper(self, nickname: str):
         })
         return whois_line.command == RPL_WHOISOPERATOR
 
+    async def _find_unset(self):
+        # check stats k (locally set kline) and g (kline set on another server)
+        # to see if anything was unset while we were gone
+        klines        = await self.database.list_klines()
+        active_klines = list()
+        await self.send(build("STATS", ["k"]))
+
+        stats_kline = Response(RPL_STATSKLINE, [SELF, ANY])
+        stats_end   = Response(RPL_ENDOFSTATS, [SELF, ANY])
+        #:lithium.libera.chat 216 sandcat sandcat :k 1.2.3.4 * * :piss off m8 (2021/8/23 20.24)|(launchd!launchd@localhost{launchd})
+        #:lithium.libera.chat 219 sandcat sandcat :k :End of /STATS report
+
+        while True:
+            stats_line = await self.wait_for({
+                stats_kline, stats_end
+            })
+            if stats_line.command == RPL_STATSKLINE:
+                active_klines.append(f"{stats_line.params[4]}@{stats_line.params[2]}")
+            else:
+                break
+
+        # do the same thing but this time with /stats g
+        await self.send(build("STATS", ["g"]))
+        while True:
+            stats_line = await self.wait_for({
+                stats_kline, stats_end
+            })
+            if stats_line.command == RPL_STATSKLINE:
+                active_klines.append(f"{stats_line.params[4]}@{stats_line.params[2]}")
+            else:
+                break
+
+        for id, mask in klines:
+            if not mask in active_klines:
```

Comment:
it'd be more performant to have `active_klines` be a set, and convert the result of `self.database.list_klines()` in to a dictionary of `{mask: id}`, and then find unset kline masks by doing `unset_klines = set(known_klines.keys()) - active_klines`

Reply:
but with this way we wouldn't have the id of the k-line that we need to remove it from the database; we'd only have the masks.

Reply:
that's why you have a dict to look the mask up in

Reply:
right...